packages/calcite-components/src/components/input-date-picker/input-date-picker.tsx

```diff
@@ -323,8 +323,10 @@ export class InputDatePicker
   }
 
   set value(value: string | string[]) {
-    const oldValue = this._value;
-    if (value !== oldValue) {
+    const valueChanged = value !== this._value;
+    const invalidValueCleared = value === "" && this.startInput?.value !== "";
```

Comment:
If the user skip setting startInput & set an invalid value in endInput value, clearing those values wouldn't work.

Reply:
Great catch. Fixed!